tests/test_jwt.py

```diff
@@ -98,5 +108,28 @@ def test_encode_with_header(self):
         self.assertEqual(jwt.header(jwt_message), header)
         self.assertEqual(decoded_payload, self.payload)
 
+    def test_allowed_algos(self):
+        secret = 'secret'
+        alg = 'HS256'
+        header = {'typ': 'urn:ietf:params:oauth:token-type:jwt', 'alg': alg}
+        # allowed_algos parameter
+        jwt_message = jwt.encode(self.payload, secret, algorithm="HS256")
+        decoded_payload = jwt.decode(jwt_message, secret, verify=True,
+                                     allowed_algos=("HS256",))
+        self.assertEqual(decoded_payload, self.payload)
+        #
+        self.assertRaises(ValueError, jwt.set_allowed_algos, 'HS256',
+                          'banana')
+        #
+        jwt.set_allowed_algos('HS256', 'RS256')
+        jwt_message = jwt.encode(self.payload, secret, algorithm="HS256")
+        decoded_payload = jwt.decode(jwt_message, secret, verify=False)
+        self.assertEqual(decoded_payload, self.payload)
+        # Disallowed algo
```

Comment:
this test would be less fragile if you used ``set_allowed_algos()`` to remove the algorithm first